File: 728-自除数.py
from typing import List


class Solution:
    def selfDividingNumbers(self, left: int, right: int) -> List[int]:
        def isselfDividing(num):
            x = num
            while x:
                x, d = divmod(x, 10)
                if d == 0 or num % d:
                    return False
            return True
        res = []
        for i in range(left, right+1):
            if isselfDividing(i):
                res.append(i)
        return res

File: 728-自除数.cpp
#include <iostream>
using namespace std;
#include <vector>
class Solution
{
public:
    vector<int> selfDividingNumbers(int left, int right)
    {
        vector<int> res;
        for (int i = left; i <= right; i++)
        {
            if (isselfDividing(i))
            {
                res.push_back(i);
            }
        }
        return res;
    }
    bool isselfDividing(int num)
    {
        int x = num;
        while (x)
        {
            int d = x % 10;
            if (d == 0 || num % d != 0)
                return false;
            x /= 10;
        }
        return true;
    }
};